From 088c70571f3fc44f6a756c42d7e034e644bcb2ce Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 30 Jul 2024 09:53:23 +0300
Subject: [PATCH] [FSW-7920] fs1: palad: release pci retry request

"Retry request" bits are set by default now, so FW must clear them
so the host can access the PCI configuration space, and other pci
resources.

Change-Id: Icc97a5d86f076172f6a6cef26dbaadc547bef2d6
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/fs1/arc/palad.c b/arch/fs1/arc/palad.c
index df14586..637d369 100644
--- a/arch/fs1/arc/palad.c
+++ b/arch/fs1/arc/palad.c
@@ -11,6 +11,7 @@
 #include <psoc_boot_conf_regs.h>
 #include <psoc_boot_conf_masks.h>
 #include <arc_regs.h>
+#include <pcie_flr_regs.h>
 #include <fs1_blocks_linux_driver.h>
 
 #include <baremetal.h>
@@ -202,6 +203,11 @@
 
 	regmap_writel(LTTSM_ENA_MASK, mmIO_0_CFG_HALF_0_CXL_EP_PCIE_AUX_BASE +
 				      mmPCIE_AUX_LTSSM_EN);
+
+	/* release request retry so the host can access pci config space */
+	regmap_writel(0, IO0_H0_PCIE_FLR(mmPCIE_FLR_APP_REQ_RETRY_EN));
+	regmap_writel(0, IO0_H0_PCIE_FLR(mmPCIE_FLR_APP_PF_REQ_RETRY_EN));
+	regmap_writel(0, IO0_H0_PCIE_FLR(mmPCIE_FLR_APP_VF_REQ_RETRY_EN));
 }
 
 
